src/ore/src/id_gen.rs

```diff
@@ -86,8 +86,9 @@ pub struct IdAllocator<A: IdAllocatorInner>(pub Arc<Mutex<A>>);
 pub trait IdAllocatorInner: std::fmt::Debug + Send {
     /// Name of the allocator.
     const NAME: &'static str;
-    /// Construct an allocator with the given range.
-    fn new(min: u32, max: u32) -> Self;
+    /// Construct an allocator with the given range. Returned ids will be OR'd with `mask`. `mask`
+    /// must not ever have any bits in the [min, max] range.
```

Comment:
I'm having trouble understanding the second sentence

>     `mask` must not ever have any bits in the [min, max] range.

So for a range of `[0, 15]`, then mask cannot have any bits set in positions 0, 1, 2, and 3. What about for a range of `[10, 15]`? Based on the `assert` below, I think the mask still cannot have any bits set in positions 0, 1, 2, and 3. So I think we actually only care about `max`, specifically `log_2(max)`. We don't want any bits set in positions <= `log_2(max)`.

Reply:
I think your description is simpler, so I'll change it to that. The original intent was "any bits that may be set by any number between min and max must not be in the mask", but that's overly complicated.